examples/v1beta1/pipelineruns/pipelinerun-with-when-expressions.yaml

```diff
@@ -0,0 +1,124 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: create-readme-file
+spec:
+  resources:
+    outputs:
+      - name: workspace
+        type: git
+  steps:
+    - name: write-new-stuff
+      image: ubuntu
+      script: 'touch $(resources.outputs.workspace.path)/README.md'
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: check-file
+spec:
+  params:
+    - name: path
+  resources:
+    inputs:
+      - name: workspace
+        type: git
+  results:
+    - name: status
+      description: indicating whether the file exists
+  steps:
+    - name: check-file
+      image: alpine
+      script: |
+        if test -f $(resources.inputs.workspace.path)/$(params.path); then
+          printf exists | tee /tekton/results/status
+        else
+          printf missing | tee /tekton/results/status
+        fi
+---
+apiVersion: tekton.dev/v1alpha1
+kind: PipelineResource
```

Comment:
If we can please replace git resource with [git-clone](https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.2) task from catalog.

Reply:
it was based on [the conditional pipelinerun example](https://github.com/tektoncd/pipeline/blob/master/examples/v1beta1/pipelineruns/conditional-pipelinerun.yaml), will update to use git-clone, thank you